app/models/concerns/category_hashtag.rb

```diff
@@ -35,6 +35,8 @@ def query_loaded_from_slugs(category_slugs, categories)
         .map(&:downcase)
         .map do |slug|
           slug_path = slug.split(":")
+          next if slug_path.empty? || slug_path.size > 2
```

Comment:
These specs actually caught a bug too which is fixed here.

Reply:
Since this is also used in `#query_from_hashtag_slug` on L13, maybe it deserves its own method? I would love to see slug promoted to its own data class, but that's probably out of scope for this PR. 🙂

Reply:
Sure, will extract